superset/viz.py

```diff
@@ -303,19 +308,30 @@ def process_query_filters(self) -> None:
         merge_extra_filters(self.form_data)
         utils.split_adhoc_filters_into_base_filters(self.form_data)
 
+    @staticmethod
+    def dedup_columns(*columns_args: Optional[List[Column]]) -> List[Column]:
+        # dedup groupby and columns while preserving order
+        labels: List[str] = []
+        deduped_columns: List[Column] = []
+        for columns in columns_args:
+            for column in columns or []:
+                label = get_column_name(column)
+                if label not in labels:
+                    deduped_columns.append(column)
```

Comment:
Maybe we also need to append label in `labels` ?

Reply:
@mayurnewase do you have repro steps for this error?

Reply:
I was just going through this feature and found pivot table allows same group by and column name, but looking at this function it's not expected.
So I thought this might be incorrecct as its not really putting anything in `labels` list.
So may need to add ```labels.append(label)``` here?

Reply:
I did a quick test, and it seems the legacy pivot table does work with overlapping groupby and column names:
![image](https://user-images.githubusercontent.com/33317356/152329475-fa7954b0-6848-4d49-b6b7-98d35d54615b.png)